### PR TITLE
Break dependency of tools/lib/src from lib/src/commands/

### PR DESCRIPTION
**Pull request by mehmetf**
This is considered an anti-pattern since it creates circular dependency within this large package. We compile each command separately internally for performance.

Instead of consuming the verbose flag, we can emit stack trace as informational message which only gets printed in verbose mode.

I also added a test for printing the stack trace.

### CODE REVIEW

**Comment by zanderso** [comment]
It sounds like we may also need a test to avoid introducing these imports. See https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/general.shard/forbidden_imports_test.dart

**Comment by mehmetf** [comment]
Neat! I did not know we have one of these. I will add a test there.

**Comment by mehmetf** [comment]
Done. PTAL.

File: packages/flutter_tools/test/general.shard/forbidden_imports_test.dart [hunk 1]
```diff
@@ -9,6 +9,32 @@ import '../src/common.dart';
...
+      .map(_asFile);
+    for (File file in files) {
+      for (String line in file.readAsLinesSync()) {
+        if (line.startsWith(RegExp(r'import.*package:'))) {
+          continue;
+        }
+        if (line.startsWith(RegExp(r'import.*commands/'))
+         || line.startsWith(RegExp(r'import.*test/'))) {
+          final String relativePath = fs.path.relative(file.path, from:flutterTools);
+          fail("$relativePath imports $line. This is an antipattern, please find another way to access the information you are using.");
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/forbidden_imports_test.dart]: How about: "This import introduces a layering violation. Please find ..."

**Review by zanderso — APPROVED** [review]
lgtm w/ error message suggestion.

**Pull request merged**